docs/versioned_docs/version-v1.3.0/02-Features/02-message-handling/03-event-hubs.md

```diff
@@ -0,0 +1,325 @@
+
+> Note that the order in which the message handlers are registered is important in the message processing.
+> In the example, when a message handler above this one is registered that could also handle the message (same message type) than that handler may be chosen instead of the one with the specific filter.
+
+### Bring your own deserialization
+You can also choose to extend the built-in message deserialization with a custom deserializer to meet your needs. 
+This allows you to easily deserialize into different message formats or reuse existing (de)serialization capabilities that you already have without altering the message router. 
+
+You start by implementing an `IMessageBodySerializer`. The following example shows how an expected type can be transformed to something else. 
+The result type (in this case `SensorReadingBatch`) will then be used to check if there is an `IAzureEventHubsMessageHandler` registered for that message type.
+
+```csharp
+using Arcus.Messaging.Pumps.Abstractions.MessageHandling;
+
+public class SensorReadingBatchMessageBodySerializer : IMessageBodySerializer
+{
+    public async Task<MessageResult> DeserializeMessageAsync(string messageBody)
+    {
+        var serializer = new XmlSerializer(typeof(SensorReading[]));
+        using (var contents = new MemoryStream(Encoding.UTF8.GetBytes(messageBody)))
+        {
+            var messages = (SensorReading[]) serializer.Deserialize(contents);
+            return MessageResult.Success(new SensorReadingBatch(messages));
+        }
+    }
+}
+```
+
+The registration of these message body serializers can be done just as easily as an `IAzureEventHubsMessageHandler`:
+
+```csharp
+using Microsoft.Extensions.DependencyInjection;
+
+public class Program
+{
+    public void ConfigureServices(IServiceCollection services)
+    {
+        // Register the message body serializer in the dependency container where the dependent services will be injected.
+        services.AddEventHubsMessagePump(...)
+                .WithEventHubsMessageHandler<SensorReadingBatchMessageHandler, SensorReading>(..., messageBodySerializer: new OrderBatchMessageBodySerializer());
+
+        // Register the message body serializer  in the dependency container where the dependent services are manually injected.
+        services.AddEventHubsMessagePump(...)
+                .WithEventHubsMessageHandler(..., messageBodySerializerImplementationFactory: serviceProvider => 
+                {
+                    var logger = serviceProvider.GetService<ILogger<OrderBatchMessageHandler>>();
+                    return new SensorReadingBatchMessageHandler(logger);
+                });
+    }
+}
+```
+
+> Note that the order in which the message handlers are registered is important in the message processing.
+> In the example, when a message handler above this one is registered that could also handle the message (same message type) than that handler may be chosen instead of the one with the specific filter.
```

Comment:
```suggestion
> In the example, when a message handler above this one is registered that could also handle the message (same message type) then that handler may be chosen instead of the one with the specific filter.
```

Reply:
https://github.com/arcus-azure/arcus.messaging/pull/337/commits/540df3e21c431f63cc55f110b98d92fd628c9172